op-challenger/game/fault/preimages/direct.go

```diff
@@ -0,0 +1,59 @@
+package preimages
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/ethereum-optimism/optimism/op-challenger/game/fault/types"
+	"github.com/ethereum-optimism/optimism/op-service/txmgr"
+	ethtypes "github.com/ethereum/go-ethereum/core/types"
+	"github.com/ethereum/go-ethereum/log"
+)
+
+var _ PreimageUploader = (*DirectPreimageUploader)(nil)
+
+type PreimageOracleContract interface {
+	UpdateOracleTx(ctx context.Context, claimIdx uint64, data *types.PreimageOracleData) (txmgr.TxCandidate, error)
+}
+
+// DirectPreimageUploader uploads the provided [types.PreimageOracleData]
+// directly to the PreimageOracle contract in a single transaction.
+type DirectPreimageUploader struct {
+	log log.Logger
+
+	txMgr    txmgr.TxManager
+	contract PreimageOracleContract
+}
+
+func NewDirectPreimageUploader(logger log.Logger, txMgr txmgr.TxManager, contract PreimageOracleContract) *DirectPreimageUploader {
+	return &DirectPreimageUploader{logger, txMgr, contract}
+}
+
+func (d *DirectPreimageUploader) UploadPreimage(ctx context.Context, claimIdx uint64, data *types.PreimageOracleData) error {
+	if data != nil {
```

Comment:
Could go either way but I think I'd probably require `data` to not be `nil` here and just not call the uploader at all when there's nothing to upload. Otherwise we have to implement the nil check in every uploader we implement.  Given most of the time there isn't anything to upload it won't go unnoticed if we forget but seems neater to do the check back in the responder.